### PR TITLE
fix ListFrameworks (could return an empty element)

### LINKED ISSUE #1040: Kubescape errors when scanning with `--use-artifacts-from`
# Description

Kubescape aborts with an error when scanning with the `--use-artifacts-from` flag.

# Environment
OS: `Ubuntu 22.04` 
Version: `Your current version is: v2.0.183 [git enabled in build: true]` 
 
# Steps To Reproduce
1. Install Kubescape
2. Download artifacts by running `kubescape download artifacts`
3. Launch a scan of a file / cluster / anything by running `kubescape scan --use-artifacts-from="~/.kubescape"`
4. Observe

# Expected behavior
Kubescape scans the input with the provided artifacts.

# Actual Behavior
Kubescape exits with the following error:
```
$ ./kubescape-ubuntu-latest scan --use-artifacts-from="$HOME/.kubescape"
[info] Kubescape scanner starting
[warning] Kubernetes cluster nodes scanning is disabled. This is required to collect valuable data for certain controls. You can enable it using the --enable-host-scan flag
[info] Downloading/Loading policy definitions
[fatal] missing required input framework name
```

### PR DESCRIPTION
**Pull request by fredbi**
Signed-off-by: Frédéric BIDON <fredbi@yahoo.com>

## Overview
Listing frameworks could return an empty element when a mix of different kinds of policy files are present.

## Additional Information
I've added a unit test case, but no integration test.

## How to Test
Please refer to the test case provided with #1040 

## Related issues/PRs:
* Resolved #1040

### CODE REVIEW

**Comment by fredbi** [comment]
cc: @vladklokun @dwertent

File: core/cautils/getter/loadpolicy.go [hunk 1]
```diff
@@ -160,7 +160,7 @@ func (lp *LoadPolicy) ListFrameworks() ([]string, error) {
 			continue
 		}
 
-		if contains(frameworkNames, framework.Name) {
+		if framework.Name == "" || contains(frameworkNames, framework.Name) {
```
Comment (vladklokun) [inline core/cautils/getter/loadpolicy.go:163]: question: do we use this check as a substitute for a check that we did not manage to unmarshal a framework?

If I understand correctly, we assume that if the `Name == ""`, then the whole supposedly unmarshaled struct holds a zero value for each of its fields. Then, if the struct holds a zero value, we assume that it did not unmarshal correctly, so the contents of the file are not a framework definition, and therefore we skip it?

Is my understanding right?

**Review by vladklokun — APPROVED** [review]
LGTM :rocket:

File: core/cautils/getter/loadpolicy.go (hunk 1 above)
Comment (fredbi) [reply core/cautils/getter/loadpolicy.go:163]: It is not a substitute: if the unmarshalling errors, then we skip earlier.
Checking the name is for cases when the descriptor has been unmarshalled without error but is obviously invalid (it is not a framework, even if the JSON may be fit in a framework struct).

However, unmarshalling errors are not a good rule to decide whether the file being loaded is indeed a framework descriptor. The name check works a little better in that respect, but again, it is not perfect.

Ideally, like previously discussed with David (@dwertent), we should have a "Kind" field in those files to clearly discriminate between the various descriptors.

Comment (vladklokun) [reply core/cautils/getter/loadpolicy.go:163]: Yup, this is pretty much what I wanted to verify. Thank you!

**Pull request merged**